Sources/MongoSwift/MongoError.swift

```diff
@@ -165,19 +165,23 @@ public struct WriteConcernFailure: Codable {
     public let code: ServerErrorCode
 
     /// A human-readable string identifying write concern error.
-    public let codeName: String
+    public let codeName: String?
```

Comment:
I think its fine to just leave as-is. Just to confirm though: one of the write concern errors returned by the server as part of these tests doesn't have a codeName, and this isn't for reading errors from a spec test json file, right?

Reply:
Hah, this is actually due to a libmongoc bug ([CDRIVER-3147](https://jira.mongodb.org/browse/CDRIVER-3147)) which I reported and subsequently forgot about. In the comments you can see we even said we'd run into this when implementing transactions, lol.

Anyways, since this is a libmongoc bug and therefore likely has a shorter path to being fixed, we should probably keep this codeName non-optional. We will have to add a custom decoder initializer like we do for `WriteFailure` [here](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/MongoError.swift#L146).

Reply:
Done.